src/QsCompiler/Tests.Compiler/CallGraphTests.fs

```diff
@@ -0,0 +1,220 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.Testing
+
+open System
+open System.Collections.Generic
+open System.Collections.Immutable
+open Microsoft.Quantum.QsCompiler.DataTypes
+open Microsoft.Quantum.QsCompiler.SyntaxExtensions
+open Microsoft.Quantum.QsCompiler.SyntaxTokens
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open Microsoft.Quantum.QsCompiler.Transformations
+open Xunit
+open Xunit.Abstractions
+
+
+type CallGraphTests (output:ITestOutputHelper) =
+
+    let qualifiedName name = 
+        ("NS" |> NonNullable<string>.New, name |> NonNullable<string>.New) |> QsQualifiedName.New
+
+    let typeParameter (id : string) = 
+        let pieces = id.Split(".")
+        Assert.True(pieces.Length = 2)
+        let parent = qualifiedName pieces.[0]
+        let name = pieces.[1] |> NonNullable<string>.New
+        QsTypeParameter.New (parent, name, Null)
+    
+    let resolution (res : (QsTypeParameter * QsTypeKind<_,_,_,_>) list) =
+        res.ToImmutableDictionary((fun (tp,_) -> tp.Origin, tp.TypeName), snd >> ResolvedType.New)
+
+    let Foo  = qualifiedName "Foo"
+    let Bar  = qualifiedName "Bar"
+
+    let FooA = typeParameter "Foo.A"
+    let FooB = typeParameter "Foo.B"
+    let BarA = typeParameter "Bar.A"
+    let BarC = typeParameter "Bar.C"
+    let BazA = typeParameter "Baz.A"
+
+    static member CheckResolution (parent, expected, [<ParamArray>] resolutions) = 
+        let compareDict (d1 : ImmutableDictionary<_,_>) (d2 : ImmutableDictionary<_,_>) = 
```

Comment:
This should probably either be not a local function, or it could omit the first parameter and assume `expected` from the parent context is the first dictionary.

Reply:
sure